kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessagePublishContext.java

```diff
@@ -33,7 +34,23 @@
     private Topic topic;
     private long startTimeNs;
     private long numberOfMessages;
-    private ManagedLedger managedLedger;
+    private long baseOffset = -1L;
+
+    @Override
+    public void setMetadataFromEntryData(ByteBuf entryData) {
+        try {
+            final BrokerEntryMetadata brokerEntryMetadata = Commands.peekBrokerEntryMetadataIfExist(entryData);
+            if (brokerEntryMetadata == null) {
+                throw new IllegalStateException(
+                        "There's no BrokerEntryData, check if your broker configured brokerEntryMetadataInterceptors");
+            }
+            baseOffset = brokerEntryMetadata.getIndex() - (numberOfMessages - 1);
```

Comment:
There maybe cases that  brokerEntryMetadata enabled, but index not used, and only contains default value. we should handle this case？